tests/test_dataerror.py

```diff
@@ -0,0 +1,16 @@
+import unittest
+from unittest.mock import patch
+from models.dataerror import DataValidationError
+from models.shopcart import Shopcart
+
+
+class TestProduct(unittest.TestCase):
+	"""Data Error Model"""
+	@patch('models.shopcart.Shopcart')
+	def test_error(self, error_mock):
+		"""Test that error can be raised"""
+		error_mock.side_effect = DataValidationError()
```

Comment:
I'm not super familiar with `mock`, so I'm not sure what this test is doing exactly. What is this line doing?

`error_mock.side_effect = DataValidationError`

and how does that raise the error?

Reply:
I'm not very familiar with mocking errors either. This was just a super simple example that error can actually be raised. 

My understanding is that the shopcart model is being mocked to be a dataValidation error, but I'm not completely sure is working the way it's supposed to.
So I @patch('models.shopcart.Shopcart') and then I tell it to have a DataValidationERROR as side effect 
And finally checks if it was raised.

It was actually missing a line where I called the model I'm not sure why it worked before.

I'm not entirely sure I'm doing things fine here. Any feedback is appreciated.

Reply:
Yeah, this test doesn't really make sense. It looks like you want to test the `Exception`. Here you're just throwing it randomly. You're already testing it in applicable spots in other tests.

If you want to test the DataValidationError, you could just ensure it extends from Exception. Otherwise, we can just close this PR and not merge.

Reply:
I agree, I'll close it.